src/Microsoft.AspNet.Authentication/AuthenticationHandler.cs

```diff
@@ -189,12 +189,12 @@ protected Task<AuthenticationTicket> AuthenticateOnceAsync()
             }
         }
 
-        protected abstract Task<AuthenticationTicket> AuthenticateAsync();
+        protected abstract Task<AuthenticationTicket> HandleAuthenticateAsync();
 
         public bool ShouldHandleScheme(string authenticationScheme)
         {
             return string.Equals(BaseOptions.AuthenticationScheme, authenticationScheme, StringComparison.Ordinal) ||
-                (BaseOptions.AutomaticAuthentication && string.IsNullOrWhiteSpace(authenticationScheme));
+                (BaseOptions.AutomaticAuthentication && authenticationScheme == string.Empty);
```

Comment:
I could have sworn there was a dot net design guideline for treating string.Empty and null as the same, but I can't find a reference for it.